src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -130,18 +130,28 @@ private static class NuGetAbstraction
         {
             public static SdkResult GetSdkResult(SdkReference sdk, object nuGetVersion, SdkResolverContext context, SdkResultFactory factory)
             {
+                var logger = new NuGetSdkLogger(context.Logger);
+
                 // Cast the NuGet version since the caller does not want to consume NuGet classes directly
                 var parsedSdkVersion = (NuGetVersion)nuGetVersion;
 
                 // Load NuGet settings and a path resolver
-                ISettings settings = Settings.LoadDefaultSettings(context.ProjectFilePath, configFileName: null, MachineWideSettings.Value as IMachineWideSettings, SettingsLoadContext.Value as SettingsLoadingContext);
+                ISettings settings;
+                try
+                {
+                    settings = Settings.LoadDefaultSettings(context.ProjectFilePath, configFileName: null, MachineWideSettings.Value as IMachineWideSettings, SettingsLoadContext.Value as SettingsLoadingContext);
+                }
+                catch (Exception e)
```

Comment:
should we just catch `XmlException` instead of generic exception?

Reply:
I don't think we should in this case as I want any exception thrown during configuration reading to bubble up to the user with a logged message.  The unit test causes this condition with bad XML but there are other things like access denied or other I/O exceptions that should be caught.